baselines/FedStar/README.md

```diff
@@ -0,0 +1,84 @@
+
+****Datasets:**** : Ambient Context, Speech Commands
+
+****Hardware Setup:**** : These experiments were run on a linux server with 56 CPU threads with 325 GB Ram with A10 GPU in it. Any machine with 8 CPU cores and 16 GB memory or more would be able to run it in a reasonable amount of time. Note: I have install tensorflow with GPU support but by default, the entire experiment runs on CPU-only mode. For cpu you need to replace value of gpus to None present in distribute_gpus function inside clients.py file.
+
+****Contributors:**** : Raj Parekh. [Github](https://github.com/Raj-Parekh24) [Mail](mailto:rajparekhwc@gmail.com)
+
+
+## Experimental Setup
+
+****Task:**** :warning: *_what’s the primary task that is being federated? (e.g. image classification, next-word prediction). If you have experiments for several, please list them_*
+
+****Model:**** :warning: *_provide details about the model you used in your experiments (if more than use a list). If your model is small, describing it as a table would be :100:. Some FL methods do not use an off-the-shelve model (e.g. ResNet18) instead they create your own. If this is your case, please provide a summary here and give pointers to where in the paper (e.g. Appendix B.4) is detailed._*
+
+****Dataset:**** :warning: *_Earlier you listed already the datasets that your baseline uses. Now you should include a breakdown of the details about each of them. Please include information about: how the dataset is partitioned (e.g. LDA with alpha 0.1 as default and all clients have the same number of training examples; or each client gets assigned a different number of samples following a power-law distribution with each client only instances of 2 classes)? if  your dataset is naturally partitioned just state “naturally partitioned”; how many partitions there are (i.e. how many clients)? Please include this an all information relevant about the dataset and its partitioning into a table._*
+
+****Training Hyperparameters:**** :warning: *_Include a table with all the main hyperparameters in your baseline. Please show them with their default value._*
+
+
+## Environment Setup
+
+:warning: _The Python environment for all baselines should follow these guidelines in the `EXTENDED_README`. Specify the steps to create and activate your environment. If there are any external system-wide requirements, please include instructions for them too. These instructions should be comprehensive enough so anyone can run them (if non standard, describe them step-by-step)._
+
+
+## Running the Experiments
+#### Set python version
+pyenv local 3.10.6
+#### Tell poetry to use python 3.10
+poetry env use 3.10.6
+#### Now install the environment
+poetry install
+#### Start the shell to activate your environment.
+poetry shell
+#### Shell script to setup dataset
+. setup_project.sh
+
+# navigate to the baselines/FedStar/FedStar
+python server.py
+
+python clients.py
```

Comment:
Your code should be executed as follows (just like other baselines do):
```
python -m fedstar.server

python -m fedstar.client
```

This will only be possible once you have (1) correctly renamed your directories to `fedstar` and use `imports` correctly in your code (see my suggestion in `server.py` about the imports).

Reply:
thanks for the suggestion, have updated the files and readme accordingly.